### ISSUE TITLE
OpenLDAP memberOf does not get detected as operational attribute

### ISSUE THREAD

**Issue by Sohalt** [cannatag/ldap3#760]
I'm using the openldap docker image from https://github.com/osixia/docker-openldap with the following ldap version:
```
(#) $OpenLDAP: slapd  (Feb  9 2019 17:02:42) $
	Debian OpenLDAP Maintainers <pkg-openldap-devel@lists.alioth.debian.org>
```
When I try to make an inetOrgPerson entry writable I get 
`ldap3.core.exceptions.LDAPCursorError: attribute 'memberOf' not in object class 'inetOrgPerson' for entry …` similar to #311. This is because the check in https://github.com/cannatag/ldap3/blob/master/ldap3/abstract/cursor.py#L248 does not see the `memberOf` attribute as an operational attribute.
If I query the servers schema, I get the following attributeTypes definition:
```
attributeTypes: ( 1.2.840.113556.1.2.102 NAME 'memberOf' DESC 'Group that the
 entry belongs to' EQUALITY distinguishedNameMatch SYNTAX 1.3.6.1.4.1.1466.115
 .121.1.12 USAGE dSAOperation X-ORIGIN 'iPlanet Delegated Administrator' )
```
Is this behavior expected and should I figure out how to adjust the ldap schema or can the library be fixed to accommodate this case?

**Comment by cannatag** [comment]
Hi, can you try to change “dsaOperation” to “dSAOperation” at line 71 of the ldap3/protocol/rfc4512.py file and let me know if this fixes your problem?

Thanks,
Giovanni

**Comment by Sohalt** [comment]
Hello,
thanks for your suggestion. It fixes the specific error I was getting, but now I am running into another error:
```
KeyError: 'memberof'


  File "/usr/local/lib/python3.7/site-packages/ldap3/abstract/entry.py", line 418, in entry_writable
    old_attr_def = writable_entry.entry_definition._attributes[original_attr.name]
  File "/usr/local/lib/python3.7/site-packages/ldap3/utils/ciDict.py", line 188, in __getitem__
    return CaseInsensitiveDict.__getitem__(self, self._aliases[self._ci_key(key)])
```

**Comment by cannatag** [comment]
Hi, do you still have this problem? I can't test it now with openldap. Could you try to give me some code showing the error?

**Comment by Sohalt** [comment]
Hello, sorry for the slow response. I still have the problem. I've pushed some example code here https://github.com/Sohalt/ldap3-error
The docker compose file starts an OpenLDAP server with some example entries, so you can simply do
```
$ docker-compose up
$ python test.py
```

**Comment by cannatag** [comment]
Hi, can you check the code in dev? 
Also, when you define the Reader() object please set the get_operational_parameters=True because by default the Cursor object doens't read operational attributes.

**Comment by Sohalt** [comment]
With the code in dev and `get_operational_attributes=True` I still get:
```
Traceback (most recent call last):
  File "~/ldap3/ldap3/utils/ciDict.py", line 186, in __getitem__
    return CaseInsensitiveDict.__getitem__(self, key)
  File "~/ldap3/ldap3/utils/ciDict.py", line 68, in __getitem__
    return self._store[self._case_insensitive_keymap[self._ci_key(key)]]
KeyError: 'memberof'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 23, in <module>
    u = user.entry_writable()
  File "~/ldap3/ldap3/abstract/entry.py", line 430, in entry_writable
    old_attr_def = writable_entry.entry_definition._attributes[original_attr.name]
  File "~/ldap3/ldap3/utils/ciDict.py", line 188, in __getitem__
    return CaseInsensitiveDict.__getitem__(self, self._aliases[self._ci_key(key)])
KeyError: 'memberof'
```

**Comment by cannatag** [comment]
Please, try to not assign a key to the memberof AttrDef:

person += AttrDef('memberOf')

**Comment by Sohalt** [comment]
With `person += AttrDef('memberOf')` it works.

**Comment by cannatag** [comment]
I've investigated. The reason is that the "memberOf" attribute is not returned in the "inetOrgPerson" schema definition. I think this is because memberOf is an operational attribute and so is returned only "on request" by the server. 

In the current version the Abstraction Layer, when converting a readable entry to a writable one needs to read the object class definition from the schema if you use a "key" instead of the real attribute name, so he doesn't find the attribute. If you use the real name it doens't need to.

I'm not sure if I can modify the Abstraction layer to allow keys for operational attributes. Will work on this later.

**Comment by cannatag** [comment]
Fixed in ldap3 v2.7.
